### DISCUSSION TITLE
The proxy tunnel request to proxy failed with status code '407'

### DISCUSSION THREAD

**Discussion by DinCR** [pnp/pnpcore#1135] [Q&A] [answered]
Hi , 
I am getting the above error when trying to access SharePoint using a .Net 7.0 console app.
host.Services.CreateScope() is triggering the erro.

Could you please help ?
Thanks

**Comment by jansenbe** [comment]
@DinCR : seems your company is forcing you to send traffic via it's proxy servers and requires each request to authenticate against that proxy server (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/407). The .NET HttpClient which we use in PnP Core SDK can be configured to use a proxy server via environment variables (see https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.defaultproxy?view=net-7.0). Can that not be an option for you?

**Comment by DinCR** [comment, marked as answer]
hi, Thank you very nuch . I got it fixed by adding the following line;
HttpClient.DefaultProxy.Credentials = System.Net.CredentialCache.DefaultCredentials;
